### PR TITLE
[Feature] Pose tracker C/C++/Python API&demos

### PR DESCRIPTION
**Pull request by lzhangzz**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: csrc/mmdeploy/apis/c/mmdeploy/pose_tracker.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct mmdeploy_pose_tracker* mmdeploy_pose_tracker_t;
+typedef struct mmdeploy_pose_tracker_state* mmdeploy_pose_tracker_state_t;
+
+typedef struct mmdeploy_pose_tracker_param_t {
+  int32_t det_interval;
```
Comment (lvhan028) [inline csrc/mmdeploy/apis/c/mmdeploy/pose_tracker.h:19]: I think it's necessary to add comments for the parameters.

File: csrc/mmdeploy/codebase/mmpose/pose_tracker/pipeline.cpp [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  Value::Array bbox_array;
+  Value track_ids_array;
+  // attach bboxes to image data
+  for (auto& bbox : bboxes) {
+    cv::Rect rect(cv::Rect2f(cv::Point2f(bbox[0], bbox[1]), cv::Point2f(bbox[2], bbox[3])));
+    bbox_array.push_back({
+        {"img", data["img"]},                                 // img
+        {"bbox", {rect.x, rect.y, rect.width, rect.height}},  // bbox
+        {"rotation", 0.f}                                     // rotation
```
Comment (lvhan028) [inline csrc/mmdeploy/codebase/mmpose/pose_tracker/pipeline.cpp:63]: Is 'rotation' still needed?

**Review by hanrui1sensetime — COMMENTED** [review]
Please fix conflict of cmakelist first.

File: csrc/mmdeploy/codebase/mmpose/pose_tracker/utils.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      center.y + .5f * scale[1],
+  };
+}
+
+vector<std::tuple<int, int, float>> greedy_assignment(const vector<float>& scores,
+                                                      vector<int>& is_valid_row,
+                                                      vector<int>& is_valid_col, float thr);
+
+template <typename T>
+inline void suppress_non_maximum(const vector<T>& scores, const vector<float>& similarities,
```
Comment (lvhan028) [inline csrc/mmdeploy/codebase/mmpose/pose_tracker/utils.h:64]: `codebase/mmdet` also implements `nms` and `iou`, any ideas to eliminate the duplicate codes?

File: csrc/mmdeploy/codebase/mmpose/pose_tracker/utils.cpp [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      std::max(0.f, std::min(img_h, yc + .5f * h)),
+  };
+}
+
+Bbox map_bbox(const Bbox& box) {
+  Point p0(box[0], box[1]);
+  Point p1(box[2], box[3]);
+  auto c = .5f * (p0 + p1);
+  auto s = p1 - p0;
+  static constexpr std::array image_size{192.f, 256.f};
```
Comment (lvhan028) [inline csrc/mmdeploy/codebase/mmpose/pose_tracker/utils.cpp:122]: why `(192.f, 256.f)`?

Comment (lzhangzz) [reply csrc/mmdeploy/codebase/mmpose/pose_tracker/utils.cpp:122]: This function is used to get actually bbox received by the pose model for debugging purpose. The values are stored only in the model and are not relative to any of the parameters we have (so can't be inferred).

File: demo/csrc/CMakeLists.txt [hunk 5]
```diff
@@ -53,5 +52,6 @@ if (MMDEPLOY_BUILD_SDK_CXX_API)
     add_example(pose_detector cpp pose_detector)
     add_example(rotated_detector cpp rotated_detector)
     add_example(pose_detector cpp pose_tracker)
```
Comment (lvhan028) [inline demo/csrc/CMakeLists.txt:54]: cannot build `pose_tracker`

File: demo/csrc/cpp/pose_tracker.cpp [hunk 6]
```diff
@@ -1,1064 +1,179 @@
...
 
-  MMDEPLOY_INFO("frames: {}, time {} ms", frame_id, dt.count());
+Args ParseArgs(int argc, char* argv[]) {
+  if (argc < 5 || argc > 6) {
+    std::cout << R"(Usage: pose_tracker device_name det_model pose_model video [output]
+  device_name  device name for execution, e.g. "cpu", "cuda"
+  det_model    object detection model path
+  pose_model   pose estimation model path
+  video        video path or camera index
+  output       output directory, will cv::imshow if omitted)";
```
Comment (lvhan028) [inline demo/csrc/cpp/pose_tracker.cpp:167]: better to add `std::endl`

**Comment by irexyc** [comment]
Can't run pose tracker demo on windows with same model that can run successful on linux.

File: demo/python/pose_tracker.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    return True
+
+
+def main():
+    args = parse_args()
+
+    video = cv2.VideoCapture(args.video)
+
+    tracker = PoseTracker(
+        det_model=args.det_model,
```
Comment (lvhan028) [inline demo/python/pose_tracker.py:71]: mismatched keywords.
In `python/pose_tracker.cpp`, `__init__` kwargs are:
```
py::arg("det_model_path"), py::arg("pose_model_path"), py::arg("device_name"),
           py::arg("device_id")
```

**Comment by lvhan028** [comment]
isort lint error. We can fix it by upgrading `isort` to 5.11.5 in `.pre-commit-config.yaml`

**Comment by lvhan028** [comment]
> isort lint error. We can fix it by upgrading `isort` to 5.11.5 in `.pre-commit-config.yaml`

https://github.com/open-mmlab/mmdeploy/pull/1685 will fix it

**Comment by hanrui1sensetime** [comment]
LGTM on android.

**Comment by lzhangzz** [comment]
> Can't run pose tracker demo on windows with same model that can run successful on linux.

should be fixed now.

**Review by lvhan028 — APPROVED** [review]
LGTM

**Review by irexyc — APPROVED** [review]

**Review by hanrui1sensetime — APPROVED** [review]

**Pull request merged**